itermocil 0.1.8

TomAnthony opened the pull request:
iTermocil allows you setup pre-configured layouts of windows and panes in iTerm2, having each open in a specified directory and execute specified commands. It is compatible with teamocil files designed for tmux sessions.

Project page: https://github.com/TomAnthony/itermocil

bfontaine commented on Library/Formula/itermocil.rb:
```diff
@@ -0,0 +1,33 @@
+class Itermocil < Formula
+  desc "iTermocil allows you setup pre-configured layouts of panes in iTerm2."
```
Please don’t include the formula’s name in its description.

bfontaine commented on Library/Formula/itermocil.rb:
```diff
@@ -0,0 +1,33 @@
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+
+    %w[PyYAML].each do |r|
```
You don’t need to use an array with `each` if you have only one resource.

TomAnthony commented:
@bfontaine I've fixed the description. I had the loop intentionally just as most Formulas need a loop and so I thought for consistency and easier updates in future I'd use one, but I've removed it for now.

bfontaine commented on Library/Formula/itermocil.rb:
```diff
@@ -0,0 +1,29 @@
+  end
+
+  test do
+    system "#{bin}/itermocil", "-h"
```
Can’t we write a test that does something more substantial than e.g. `-h`/`-v`?

bfontaine replied:
Maybe something like this would work?
```
ENV["EDITOR"] = "cat"
system "#{bin}/itermocil", "--edit", "sample"
```

TomAnthony replied:
I don't actually install any layout files as part of the install. I could check for and create a folder and put a couple of default layout files in there. Some people may not want me adding files to their setup though (though the next release of iTermocil will move to having its own layout folder [rather than sharing with teamocil] so then this will be easier). Thoughts?

bfontaine replied:
Can’t you create a layout with `itermocil`? Note that in the test environment, `$HOME` is the `testpath` so it’s ok to create files under e.g. `~/.something`.

TomAnthony replied:
It can, but the problem is the directory doesn't exist. Also I wasn't sure that opening an editor to allow you to create a layout was any better as a test than running -h.

If we did think it was better then I could get the homebrew formula to add an .itermocil directory in the users home directory, but wasn't sure if that was 'bad form'.

bfontaine replied:
Yeah you can’t do that in a Homebrew formula. In the test you can create a directory with `(testpath/"somedir").mkpath`.

TomAnthony replied:
Ok, got it. Thanks.

The best test would actually be to copy a file in then run itermocil with the --debug flag to parse the file. Can I copy a file for the test in that way?

Something like this:

```
testpath.install "test_layouts/_tiled_4_panes.yml"
ENV["EDITOR"] = "cat"
system "#{bin}/itermocil", "--debug", "--layout", "_tiled_4_panes.yml"
```

bfontaine replied:
`.install` moves the file instead of copying it, so just use something like
`cp "#{prefix}/test_layouts/_tiled_4_panes.yml", "test.yml"`

TomAnthony replied:
Done. I tested this and it seems to be working as intended. :)

bfontaine commented on Library/Formula/itermocil.rb:
```diff
@@ -0,0 +1,29 @@
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install "itermocil"
+    bin.install "itermocil.py"
```
Is there any reason to have two binaries with almost the same name?

TomAnthony replied:
For some legacy support for some users. I'll remove them in the future.

bfontaine commented:
@TomAnthony Could you please rebase your branch on top of `master`? Our CI bot has some trouble applying the latest patch.

TomAnthony commented:
@bfontaine I rebased and it looks like CI bot is happy now. :)

TomAnthony commented:
Updated the test so it actually parses a file. Also re-ran `brew audit --strict --online`. Hopefully good to go. Thanks for all the help @bfontaine.

bfontaine commented:
Seems like it fails on the bot:
```
==> /usr/local/Cellar/itermocil/0.1.8/bin/itermocil --debug --layout test.yml
26:33: execution error: File iTerm wasn’t found. (-43)
Traceback (most recent call last):
  File "/usr/local/Cellar/itermocil/0.1.8/libexec/bin/itermocil", line 6, in <module>
    main()
  File "/usr/local/Cellar/itermocil/0.1.8/libexec/bin/itermocil.py", line 687, in main
    instance = Itermocil(filepath, here=args.here, cwd=cwd)
  File "/usr/local/Cellar/itermocil/0.1.8/libexec/bin/itermocil.py", line 28, in __init__
    major_version = self.get_major_version()
  File "/usr/local/Cellar/itermocil/0.1.8/libexec/bin/itermocil.py", line 95, in get_major_version
    return float(v[:3])
ValueError: could not convert string to float: 
Error: itermocil: failed
Failed executing: /usr/local/Cellar/itermocil/0.1.8/bin/itermocil --debug --layout test.yml
```

apjanke commented:
Closing this due to inactivity/lack of response. If you'd still like to pursue this, feel free to respond to build failure and re-open on the new [`homebrew/core` repo](https://github.com/homebrew/homebrew-core).
